src/Gaufrette/Adapter/Ftp.php

```diff
@@ -15,25 +15,25 @@ class Ftp implements Adapter, FileFactory, ListKeysAware, SizeCalculator
 {
     /** @var null|resource|\FTP\Connection */
     protected $connection = null;
-    protected $directory;
-    protected $host;
-    protected $port;
-    protected $username;
-    protected $password;
-    protected $passive;
-    protected $create;
-    protected $mode;
-    protected $ssl;
-    protected $timeout;
-    protected $fileData = [];
-    protected $utf8;
+    protected string $directory;
+    protected string $host;
+    protected mixed $port;
+    protected mixed $username;
+    protected mixed $password;
+    protected mixed $passive;
+    protected mixed $create;
+    protected mixed $mode;
+    protected mixed $ssl;
+    protected mixed $timeout;
+    protected array $fileData;
```

Comment:
Why do you remove the default value ? If you create a file, it will raise an error.